### DISCUSSION TITLE
Custom properties List box

### DISCUSSION THREAD

**Discussion by jcbk101** [mapeditor/tiled#3394] [Ideas]
Hello, all. I am not sure this is supported, but I figured it might be an ok add.

Is there a way to add support for a drop-down list of strings for custom string properties?
I want to make designing this game as efficient as possible and having to type / paste what I want
in the entry box gets tedious for many objects + maps. For ref, see attached photos.

Example: Object->Type has a drop-down list that the app retrieves data for. This data is from data enter in the 'Data Types Editor' and saved to the project.
![list](https://user-images.githubusercontent.com/75145117/175838495-345061cd-bb47-4096-be83-1c080a261442.png)

Having a similar ability for string types in the 'Custom properties' section would allow for quicker flow and data entry.
![no list](https://user-images.githubusercontent.com/75145117/175838499-627286dc-ade6-457a-810a-864b8eef871d.png)

**Comment by bjorn** [comment]
This should be covered by [defining custom enums](https://doc.mapeditor.org/en/stable/manual/custom-properties/#custom-types), a feature added in Tiled 1.8. You'd then select your enum type when adding the custom property, and with the "Save as" option set to "String" there should be no changes necessary for your map loader.

**Comment by jcbk101** [comment]
Thank you.


________________________________
